rflx/specification/parser.py

```diff
@@ -460,73 +1096,239 @@ def create_proven_message(
         )
-    if package.identifier != package.end_identifier:
+    if identifier != package.f_end_identifier.text:
         error.append(
-            f'inconsistent package identifier "{package.end_identifier}"',
+            f'inconsistent package identifier "{package.f_end_identifier.text}"',
             Subsystem.PARSER,
             Severity.ERROR,
-            package.end_identifier.location,
+            node_location(package.f_end_identifier, name),
         )
         error.append(
-            f'previous identifier was "{package.identifier}"',
+            f'previous identifier was "{identifier}"',
             Subsystem.PARSER,
             Severity.INFO,
-            package.identifier.location,
+            node_location(package.f_identifier, name),
         )
-    if filename:
-        expected_filename = f"{str(package.identifier).lower()}.rflx"
-        if filename != expected_filename:
+    if origname:
+        expected_filename = f"{identifier.lower()}.rflx"
+        if origname.name != expected_filename:
             error.append(
-                f'file name does not match unit name "{package.identifier}",'
+                f'file name does not match unit name "{identifier}",'
                 f' should be "{expected_filename}"',
                 Subsystem.PARSER,
                 Severity.ERROR,
-                package.identifier.location,
+                node_location(package.f_identifier, origname),
             )
-    for t in package.types:
-        if is_builtin_type(t.identifier.name):
+    for t in package.f_declarations:
+        if isinstance(t, TypeDecl) and model.is_builtin_type(create_id(t.f_identifier, name).name):
             error.append(
-                f'illegal redefinition of built-in type "{t.identifier.name}"',
+                f'illegal redefinition of built-in type "{t.f_identifier.text}"',
                 Subsystem.MODEL,
                 Severity.ERROR,
-                t.location,
+                node_location(t, name),
             )
+
+
+class Parser:
+    def __init__(self, skip_verification: bool = False, cached: bool = False) -> None:
+        self.skip_verification = skip_verification
+        self.__specifications: OrderedDict[Path, Specification] = OrderedDict()
+        self.__types: List[model.Type] = [
+            *model.BUILTIN_TYPES.values(),
+            *model.INTERNAL_TYPES.values(),
+        ]
+        self.__sessions: List[model.Session] = []
+        self.__cache = Cache(cached)
+
+    def __convert_unit(
+        self, spec: Specification, specfile: Path = None, transitions: List[ID] = None
+    ) -> RecordFluxError:
+        transitions = transitions or []
+        error = RecordFluxError()
+
+        if spec:
+            parent = specfile.parent if specfile else Path(".")
+            filename = parent / Path(f"{spec.f_package_declaration.f_identifier.text.lower()}.rflx")
+            self.__specifications[filename] = (specfile, spec)
```

Comment:
The naming of `specfile` and `filename` doesn't make clear their different meanings. What is called `filename` here, is called `specfile` in other places. The naming should be consistent throughout the file.

Reply:
Changed in 8a4b76b6c311b75b70acceff98fc86e2f7ff160c.